lib/profiles/user-profile-data.js

```diff
@@ -0,0 +1,473 @@
+      }
+    }
+
+    function getRequestingUserProfile (profileDojos, requestingUserDojos, wfCb) {
+      if (args.user) {
+        seneca.act({role: 'cd-profiles', cmd: 'load_user_profile', userId: args.user.id},
+          function (err, requestingUserProfile) {
+            if (err) return done(err);
+            return wfCb(null, profileDojos, requestingUserDojos, requestingUserProfile);
+        });
+      } else {
+        return wfCb(null, profileDojos, null, null);
+      }
+    }
+
+    function mapFlags (profileDojos, requestingUserDojos, requestingUserProfile, wfCb) {
+      var requestingUser = args.user;
+      // We look at viewer rights
+      _.each(requestingUserDojos, function (requestingUserDojo) {
+        if (_.contains(requestingUserDojo.userTypes, 'champion')) flags.requestingUser.isChampionOf = true;
+        if (_.contains(requestingUserDojo.userTypes, 'mentor')) flags.requestingUser.isMentorOf = true;
+        if (_.find(requestingUserDojo.userPermissions, {'title':'Dojo Admin','name':'dojo-admin'})) flags.requestingUser.isDojoAdminOf = true;
+        if (_.find(requestingUserDojo.userPermissions, {'title':'Ticketing Admin','name':'ticketing-admin'})) flags.requestingUser.isTicketingAdminOf = true;
+      });
+      // Viewed user flags
+      _.each(profileDojos, function (profileDojo) {
+        if (_.contains(profileDojo.userTypes, 'champion')) flags.user.isChampion = true;
+        if (_.find(profileDojo.userPermissions, {'title':'Dojo Admin','name':'dojo-admin'})) flags.user.isDojoAdmin = true;
+        if (_.find(profileDojo.userPermissions, {'title':'Ticketing Admin','name':'ticketing-admin'})) flags.user.isTicketingAdmin = true;
+      });
+      // We look at relationship between user and viewer
+      _.find(requestingUserDojos, function (requestingUserDojo) {
+        var sharedProfileDojos = _.filter(profileDojos, function (profileDojo) {
+          return profileDojo.dojoId === requestingUserDojo.dojoId;
+        });
+        _.each(sharedProfileDojos, function (sharedProfileDojo) {
+          if (_.find(sharedProfileDojo.userPermissions, {'title':'Dojo Admin','name':'dojo-admin'})) flags.shared.isDojoAdmin = true;
+          if (_.find(sharedProfileDojo.userPermissions, {'title':'Ticketing Admin','name':'ticketing-admin'})) flags.shared.isTicketingAdmin = true;
+        });
+      });
+
+      flags.shared.isParentOf = _.includes(profile.parents, requestingUser.id)
+      flags.shared.isChildrenOf = _.includes(profile.children, requestingUser.id);
+
+      wfCb(null, profile, requestingUserProfile);
+    }
+
+    function filterFields (err, profile, requestingUserProfile) {
+      var allowedFields = [];
+
+      if (_.contains(profile.userTypes, 'attendee-o13')) {
+        allowedFields = _.union(allowedFields, allowedOptionalFields['attendee-o13']);
+      }
+
+      if (_.contains(profile.userTypes, 'champion')) {
+        allowedFields = _.union(allowedFields, allowedOptionalFields['champion']);
```

Comment:
@rosalanghammer If we have an attendee-o13 who's also a champion (Youth Champion), should their profile be as open as an adult champion, or still restricted as an attendee-o13?